src/pipes/decode-params-pipe.ts

```diff
@@ -0,0 +1,49 @@
+import { PipeTransform } from "@nestjs/common";
+import { ZodObject, ZodRawShape } from "zod";
+import { zodToJsonSchema } from "zod-to-json-schema";
+
+type JsonSchema =
+  | {
+      type: "string" | "number";
+    }
+  | {
+      type: "array";
+      items: {
+        type: "string" | "number";
+      };
+    };
+
+export class DecodeParamsPipe<T extends ZodRawShape> implements PipeTransform {
```

Comment:
The code here lgtm but I think it could do with some commenting so that someone looking at this code for the first time understands what it's doing and why it's necessary. You could add it at the top of the class, or inside the `transform` method, or a combination of both.

Reply:
Wilco